components/script/textinput.rs

```diff
@@ -1047,30 +1140,48 @@ impl<T: ClipboardProvider> TextInput<T> {
             .otherwise(|| {
                 if let Key::Character(ref c) = key {
                     self.insert_string(c.as_str());
-                    return KeyReaction::DispatchInput;
+                    return KeyReaction::DispatchInput(
+                        Some(c.to_string()),
+                        IsComposing::NotComposing,
+                        InputType::InsertText,
+                    );
                 }
                 if matches!(key, Key::Named(NamedKey::Process)) {
-                    return KeyReaction::DispatchInput;
+                    return KeyReaction::DispatchInput(
+                        None,
+                        IsComposing::NotComposing,
```

Comment:
Since the Process key is part of using an IME, I think we this event to be composing here: `true if this event is dispatched during a [dead key](https://w3c.github.io/uievents/#keys-dead) sequence or while an [input method editor](https://w3c.github.io/uievents/#input-method-editor) is active (such that [composition events](https://w3c.github.io/uievents/#events-compositionevents) are being dispatched)`